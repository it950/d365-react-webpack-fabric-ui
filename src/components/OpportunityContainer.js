import React, { Component } from 'react'
import { DefaultButton } from 'office-ui-fabric-react/lib/Button'
import { Spinner, SpinnerSize } from 'office-ui-fabric-react/lib/Spinner'

import OpportunityList from './OpportunityList'
import OpportunityPanel from './OpportunityPanel'
import opportunities from '../../db.js'

class App extends Component {
  state = {
    showPanel: false,
    opportunities: [],
  }

  componentDidMount() {
    this.loadOpportunities()
  }

  loadOpportunities() {
    /*eslint no-undef: 0*/
    // const url = Xrm.Page.context.prependOrgName(
    //   '/api/data/v8.0/opportunities?$filter=salesstage eq 0&$orderby=createdon desc' +
    //     '&$top=10' +
    //     '&$select=opportunityid,name,createdon',
    // )

    // const response = fetch(url, {
    //   credentials: 'same-origin',
    // })

    // fetch(url, {
    //   credentials: 'same-origin',
    // })
    //   .then(res => res.json())
    //   .then(json => this.setState({ opportunities: json.value }))

    setTimeout(() => this.setState({ opportunities }), 1000)
  }

  handleClosePanel = () => {
    this.setState({ showPanel: false })
  }

  handleShowPanel = () => {
    this.setState({ showPanel: true })
  }

  render() {
    const { showPanel, opportunities } = this.state
    return (
      <div>
        <h1>Opportunity Summary</h1>
        {opportunities.length > 0 ? (
          <div>
            <OpportunityList opportunities={this.state.opportunities} />
            <DefaultButton
              description="Opens the Panel"
              onClick={this.handleShowPanel}
              text="Open Panel"
            />
            <OpportunityPanel
              isOpen={showPanel}
              onClosePanel={this.handleClosePanel}
            />
          </div>
        ) : (
          <Spinner
            size={SpinnerSize.large}
            label="Loading opportunities..."
            ariaLive="assertive"
          />
        )}
      </div>
    )
  }
}

export default App
