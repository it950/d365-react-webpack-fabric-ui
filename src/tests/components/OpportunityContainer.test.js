import React from 'react'
import { shallow } from 'enzyme'

import OpportunityContainer from '../../components/OpportunityContainer'

describe('OpportunityContainer', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<OpportunityContainer />)
    expect(wrapper).toMatchSnapshot()
  })
})
