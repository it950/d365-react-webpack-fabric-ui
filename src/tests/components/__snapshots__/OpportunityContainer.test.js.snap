// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpportunityContainer should render correctly 1`] = `
<div>
  <h1>
    Opportunity Summary
  </h1>
  <Spinner
    ariaLive="assertive"
    label="Loading opportunities..."
    size={3}
  />
</div>
`;
